proto/frequenz/api/reporting/v1/reporting.proto

```diff
@@ -0,0 +1,534 @@
+// MIT
+
+syntax = "proto3";
+
+package frequenz.api.reporting.v1;
+
+import "google/protobuf/timestamp.proto";
+
+import "frequenz/api/common/v1/components.proto";
+import "frequenz/api/common/v1/components/data.proto";
+import "frequenz/api/common/v1/metrics.proto";
+import "frequenz/api/common/v1/microgrids.proto";
+import "frequenz/api/common/v1/pagination.proto";
+
+// The Reporting service provides services for real-time and historical metrics monitoring
+// of various microgrid components like Batteries, EV Chargers, and Inverters.
+//
+// Utilize these APIs for tasks ranging from real-time metric streaming to complex,
+// formula-driven historical data aggregation. For detailed information on component
+// categories and the metrics they provide, please consult the referenced component categories
+// imported above.
+//
+service Reporting {
+  // Fetches historical metrics for a list of microgrid components.
+  //
+  // !!! note
+  //     This RPC endpoint retrieves historical metrics data for components of one or more
+  //     microgrids. Clients need to provide at least one microgrid ID and one component ID
+  //     to get a result.
+  rpc ListMicrogridComponentsData(ListMicrogridComponentsDataRequest)
+    returns (ListMicrogridComponentsDataResponse) {}
+
+  // Streams metrics for a list of microgrid components.
+  //
+  // !!! note
+  //     This RPC endpoint streams metrics data for components of one or more microgrids.
+  //     Clients need to provide at least one component ID to get a result.
+  rpc ReceiveMicrogridComponentsDataStream(ReceiveMicrogridComponentsDataStreamRequest)
+    returns (stream ReceiveMicrogridComponentsDataStreamResponse) {}
+
+  // Fetches aggregated historical metrics based on user-defined formulas for a list of
+  // microgrid components.
+  //
+  // !!! note
+  //     This RPC endpoint retrieves aggregated historical metrics data based on user-defined
+  //     formulas for microgrid components. At least one formula must be provided for a valid query.
+  rpc ListAggregatedMicrogridComponentsData(ListAggregatedMicrogridComponentsDataRequest)
+    returns (ListAggregatedMicrogridComponentsDataResponse) {}
+
+  // Streams aggregated metrics based on user-defined formulas for a list of microgrid
+  // components.
+  //
+  // !!! note
+  //     This RPC endpoint retrieves aggregated historical metrics data based on user-defined
+  //     formulas for microgrid components. At least one formula for one microgrid must be provided
+  //     for a valid result.
+  rpc ReceiveAggregatedMicrogridComponentsDataStream(ReceiveAggregatedMicrogridComponentsDataStreamRequest)
+    returns (ReceiveAggregatedMicrogridComponentsDataStreamResponse) {}
+}
+
+// Time-based filter for querying aggregated microgrid components data.
+//
+// !!! note
+//     Specifies the start and end time for the query. Both fields are optional, but
+//     their absence or presence has specific implications for the query results.
```

Comment:
Does the query include the start and end timestamp? Does this depend on the bounds option?

Reply:
Okay. Are the start and end timestamps then included in the output?

Reply:
Since this API is about reading time-series data, the output will be a collection of rows, and each row will have a timestamp. These output timestamps will be >= start and <= end.
If this is not clear, then we can add this to the docs.